src/Color.cc

```diff
@@ -187,13 +188,28 @@ void Color::SetFromYUV(const float _y, const float _u, const float _v)
 }
 
 //////////////////////////////////////////////////
-float Color::operator[](const unsigned int _index)
+float& Color::operator[](const unsigned int _index)
 {
-  return (*static_cast<const Color *>(this))[_index];
+  switch (_index)
+  {
+    case 0:
+      return this->r;
+    case 1:
+      return this->g;
+    case 2:
+      return this->b;
+    case 3:
+      return this->a;
+    default:
+      break;
+  }
+
+  std::cerr << "Trying to read index " << _index << " of Color" << std::endl;
+  throw std::runtime_error("Index Error: Color index out of range");
```

Comment:
We don't typically use exceptions in gz-math, so please remove this. But I do see the problem with what we should do if the _index is out of range since we can return the `NAN_F` by reference.  The best thing I can think of is to create a static float here and return that by value.

```suggestion
  static float invalidValue = NAN_F;
  return invalidValue;
```